### PR TITLE
Refactor Classification Tasks

### PR DESCRIPTION
**Pull request by isaaccorley**
- Adds `num_classes` as arg to `ClassificationTask` and `MultiLabelClassificationTask`
- Removes unnecessary tasks
- Updates configs

### CODE REVIEW

File: tests/trainers/test_so2sat.py [hunk 1]
```diff
@@ -44,6 +44,7 @@ def config(self, request: SubRequest, bands: Tuple[str, int]) -> Dict[str, Any]:
         task_args["in_channels"] = bands[1]
         loss, weights = request.param
         task_args["loss"] = loss
+        task_args["num_classes"] = 17
```
Comment (adamjstewart) [inline tests/trainers/test_so2sat.py:47]: Is this needed now that the default configs have been updated?

**Review by adamjstewart — COMMENTED** [review]
This looks even better than I imagined! One quick comment and then I think this is ready to merge.

File: tests/trainers/test_so2sat.py (hunk 1 above)
Comment (isaaccorley) [reply tests/trainers/test_so2sat.py:47]: Good question, it's not. I'll remove. Can you add it as a suggestion so I can do it from my phone lol.

Comment (adamjstewart) [reply tests/trainers/test_so2sat.py:47]: I'm also on my phone lol

**Review by calebrob6 — APPROVED** [review]
I don't know how to suggest, but edited directly

**Pull request merged**